services/dynamic-scheduler/tests/conftest.py

```diff
@@ -50,32 +50,13 @@ def docker_compose_service_dynamic_scheduler_env_vars(
 ) -> EnvVarsDict:
     """env vars injected at the docker-compose"""
 
-    dynamic_scheduler = yaml.safe_load(services_docker_compose_file.read_text())[
-        "services"
-    ]["dynamic-schdlr"]
-
-    def _substitute(item):
-        key, value = item.split("=")
-        if m := re.match(r"\${([^{}:-]\w+)", value):
-            expected_env_var = m.group(1)
-            try:
-                # NOTE: if this raises, then the RHS env-vars in the docker-compose are
-                # not defined in the env-devel
-                if value := env_devel_dict[expected_env_var]:
-                    return key, value
-            except KeyError:
-                pytest.fail(
-                    f"{expected_env_var} is not defined in .env-devel but used in docker-compose"
-                    f" services[{dynamic_scheduler}].environment[{key}]"
-                )
-        return None
-
-    envs: EnvVarsDict = {}
-    for item in dynamic_scheduler.get("environment", []):
-        if found := _substitute(item):
-            key, value = found
-            envs[key] = value
```

Comment:
Q: are you sure the new code is equivalent to this?

If you remove and env var from the .env-devl file what happens? Will you get an error here? The previous error message was very on point and would point you to what is actually wrong, instead of having a fail way down the line in the tests.

Reply:
You are right. I corrected it. thx